### ISSUE TITLE
make fail

### ISSUE THREAD

**Issue by katamari1992** [madsciencecoder/Toontown-Controller#2]
I am attempting to build on ubuntu xenial and it is failing to do so
I am getting this error
https://paste2.org/8gZKa4Kn
I tried creating the folder/file that it desires but it just created different issues can I get pointed in the correct direction 
Also unsure if I have all the dependancies

**Comment by ekureina** [comment]
you probably need to git clone --recursive to get the library to pull in

**Comment by katamari1992** [comment]
I did that

**Comment by madsciencecoder** [comment]
Hello, sorry for taking a while to respond.  From what I can tell, the library file you are missing is provided by the libxcb-keysyms1-dev package in Ubuntu.  Once you install that you might need to run qmake again to re-generate the makefile.

I don't think I ever figured out all the dependency packages for Ubuntu so you might run into other missing libraries but a quick google search should provide what packages provide them.

**Comment by katamari1992** [comment]
yay new error!
https://paste2.org/cwIphedU
I'm searching myself but there's a lot of code that I don't really understand

**Comment by katamari1992** [comment]
From what little I understand of code and the errors I am receiving I think it's a problem in the code
The error is saying that there is no function to call

src/controller.cpp:52:6: error: no matching function for call to ‘Controller::connect(UGlobalHotkeys*&, void (UGlobalHotkeys::*)(size_t), Controller::Controller(KeyMap*, QWidget*)::<lambda(size_t)>)’

My suspicsion is that global hotkeys got updated and the controller didn't.

**Comment by madsciencecoder** [comment]
I don't believe uglobalhotkeys has been updated, and git submodules checkout the specific version I used when I made the program.  If I were to guess, Qt itself changed something that made it incompatible.

I don't have a lot of time anymore to work on personal projects and I haven't use this program in a while myself so I'm not sure if I'll have interest in updating it.  I'll try to look into it sometime this weekend to see if it's something that can be easily fixed.

**Comment by katamari1992** [comment]
I apologize for not being able to be more useful in matters like this.
I've tried programming before and found it well beyond me.
